### PR TITLE
ENG-9882 remove client dependency from the http interface

### PR DESCRIPTION
**Pull request by vtkstef**

### CODE REVIEW

File: src/frontend/org/voltdb/AuthSystem.java [hunk 1]
```diff
@@ -599,6 +615,85 @@ public boolean isAuthEnabled() {
...
+                return PERMS.contains(p[0]);
+            } else if (p == null || p.length == 0) {
+                return false;
+            } else {
+                return PERMS.containsAll(Arrays.asList(p));
+            }
+        }
+
+        @Override
+        public boolean authorizeConnector(String connectorName) {
```
Comment (akhanzode) [inline src/frontend/org/voltdb/AuthSystem.java]: Is this used/needed? Do AuthDisabledUser implement this? Can you document what connectorName would translate to?

File: src/frontend/org/voltdb/AuthSystem.java [hunk 2]
```diff
@@ -599,6 +615,85 @@ public boolean isAuthEnabled() {
...
```
Comment (akhanzode) [inline src/frontend/org/voltdb/AuthSystem.java]: same as above.

File: src/frontend/org/voltdb/HTTPClientInterface.java [hunk 3]
```diff
@@ -141,7 +132,7 @@ public HTTPClientInterface() {
         final ClientResponseImpl r = new ClientResponseImpl(ClientResponse.CONNECTION_TIMEOUT,
                 new VoltTable[0], "Request Timeout");
         m_timeoutResponse = r.toJSONString();
-        m_closeAllExecutor = CoreUtils.getSingleThreadExecutor("HttpClientInterface-closeAll");
+        m_closeAllExecutor = CoreUtils.getSingleThreadExecutor("HttpClientInterface-closeAll"); // STEBUG remove
```
Comment (akhanzode) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: Do u mean to leave STEBUG?

File: src/frontend/org/voltdb/HTTPClientInterface.java [hunk 4]
```diff
@@ -331,20 +312,19 @@ public void process(Request request, HttpServletResponse response) {
...
                 continuation.complete();
             }
             request.setHandled(true);
-        } finally {
-            releaseClient(authResult, forceClose);
         }
     }
 
+    public boolean callProcedure(final AuthenticationResult ar, int timeout, ProcedureCallback cb, String procName, Object...args) {
+        return m_invocationHandler.get().callProcedure(ar.m_authUser, ar.m_adminMode, timeout, cb, procName, args);
```
Comment (akhanzode) [inline src/frontend/org/voltdb/HTTPClientInterface.java]: inconsistent formatting?

File: src/frontend/org/voltdb/InternalAdapterTaskAttributes.java [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    final boolean m_isAdmin;
+    final long m_connectionId;
+    final String m_name;
+    final InternalConnectionContext m_proxy;
+
+    public InternalAdapterTaskAttributes(String adapterName, boolean isAdmin, long connectionId) {
+        m_isAdmin = isAdmin;
+        m_connectionId = connectionId;
+        m_name = adapterName;
+        m_proxy = null;
```
Comment (akhanzode) [inline src/frontend/org/voltdb/InternalAdapterTaskAttributes.java]: Why not have a dummy proxy when no backpressure is handled and use just one constructor?

File: src/frontend/org/voltdb/InternalClientResponseAdapter.java [hunk 6]
```diff
@@ -131,7 +139,7 @@ public int getPartitionId() {
 
         @Override
         public InternalConnectionContext getInternalContext() {
-            return m_context;
+            return m_kattrs;
```
Comment (akhanzode) [inline src/frontend/org/voltdb/InternalClientResponseAdapter.java]: maybe better getter name. You are returning InternalAdaptorTaskAttributes but getter says internal context.

File: src/frontend/org/voltdb/RealVoltDB.java [hunk 7]
```diff
@@ -2232,29 +2232,21 @@ else if (m_catalogContext.catalogVersion != expectedCatalogVersion) {
...
-            // 3. update HTTPClientInterface (asynchronously)
-            // This purges cached connection state so that access with
-            // stale auth info is prevented.
-            if (m_adminListener != null)
-            {
-                m_adminListener.notifyOfCatalogUpdate();
-            }
-
-            // 4. Flush StatisticsAgent old catalog statistics.
+            // 3. Flush StatisticsAgent old catalog statistics.
```
Comment (akhanzode) [inline src/frontend/org/voltdb/RealVoltDB.java]: if there is no notification on UAC how does one disable json interface or for that matter change port?

**Comment by vtkstef** [comment]
--squash merger

**Pull request closed without merging**